src/wallet/wallet2.cpp

```diff
@@ -3153,6 +3153,60 @@ bool wallet2::verify_password(const std::string& keys_file_name, const epee::wip
   return r;
 }
 
+/*!
+ * \brief determine the key storage for the specified wallet file
+ * \param keys_file_name  Keys file to verify password for
+ * \param password        Password to verify
+ * \return                -1: incorrect password, 0 = default hw, 1 ledger hw
```

Comment:
That seems not very nice. How about returning an error code and have a parameter to return the result, which would not be 1 for ledger, but something like a name or an enum somewhere (I assume you'd choose 2 for trezor, etc).

Reply:
yes. style question: where would i put the return parameter in the argument list? note that the last argument right now is an optional ```uint64_t```?

Reply:
I don't really care as long as it's sensible. I'm just a bit weirded out by the return type being so ad hoc magic numbers.